src/include/codegen/aggregation.h

```diff
@@ -71,46 +95,59 @@ class Aggregation {
   bool IsGlobal() const { return is_global_; }
 
   //===--------------------------------------------------------------------===//
-  // Little struct to map the aggregates we physically store to the higher level
-  // aggregates. It is possible that the number of these structs is greater than
-  // the total number of aggregates the caller has setup. This can occur for two
-  // reasons:
+  // Little struct to map higher level aggregates to their physical storage
+  // and to their hash tables if they are distinct.
   //
-  // 1) Some aggregates decompose into multiple aggregations. For example, AVG()
-  //    aggregates decompose into a SUM() and COUNT(), therefore occupying three
-  //    slots: one each for the sum, count, and logical average.
-  // 2) There are occasions where components of aggregates can be shared across
-  //    multiple aggregates.  An example is a SUM(a) and AVG(a). Both of these
-  //    will share the summation aggregate on 'a'.
+  // Some aggregates decompose into multiple components. For example, AVG()
+  // aggregates decompose into a SUM() and COUNT(). Therefore, the storage
+  // indexes are stored in an array. The array has fixed size of the maximum
+  // number of components that a aggregation is decomposed to, so for now
+  // only 2 for AVG. The aggregations have to know which component is
+  // stored at which index.
   //
   // Storing the mapping from the physical position the aggregate is stored to
   // where the caller expects them allows us to rearrange positions without
   // the caller knowing or caring.
   //===--------------------------------------------------------------------===//
-  struct AggregateInfo {
-    // The type of aggregate
-    ExpressionType aggregate_type;
+  static const unsigned int max_comp_size_ = 2;
```

Comment:
For static constants, we use the camel-casing formats. Rename this to `kMaxNumComponents`